basic_module/distribution/src/impls.rs

```diff
@@ -100,3 +100,15 @@ impl CloseBlockHandler for Distribution {
         Ok(())
     }
 }
+
+impl FeeManager for Distribution {
```

Comment:
I don't think the struct `Distribution` is not responsible for the `FeeManager` functionality. `Distribution` contains the internal state `block_number` and `author` to remember in what context it is executing transactions. However, `FeeManager` can achieve its goal regardless of the state of `Distribution`. In the sense that it will be instantiated and linked to another module, the variable in this instance will never be used. Therefore, it's better to implement `FeeManager` trait with another struct.